lib/route/ROUTING.md

```diff
@@ -0,0 +1,446 @@
+# Obelisk Routing
+
+The `obelisk-routes` package is designed to help with managing the paths and parameters for
+routing in your application.
+
+This is distinct from packages like `servant` that are designed for specifying endpoints for a
+REST API. When building routes with `obelisk-routes` you're not going to be handling headers or
+HTTP methods.
```

Comment:
Backend route handlers do need to think about HTTP methods/codes. But it's not strictly part of routing system. So I guess this is fine but it's maybe a bit surprising to a "normie" understanding of routing.

Reply:
We should provide the reason here, rather than just stating the fact, because most readers will probably think it's weird.  The reason is that our current incarnation of routes is designed to support only features that are usable on *both* the frontend and the backend.  That's because there are some laws we're trying to preserve here, like: backend pre-rendering produces a page that looks the same as what the frontend will render.  This rules out the use of the hash string, because it doesn't get sent to the backend.  (We could support the hash string if we could guarantee that the frontend doesn't use it in any way other than going to the anchor, because that's what will happen with the pre-rendered version!)

Reply:
I've overhauled the introduction to try to convey this point and a few others. Probably still rough edges...